lib/fluent/plugin/out_webhdfs.rb

```diff
@@ -100,28 +99,18 @@ def desc(description)
 
   def initialize
     super
-    require 'net/http'
-    require 'time'
-    require 'webhdfs'
-
     @compressor = nil
   end
 
-  # Define `log` method for v0.10.42 or earlier
-  unless method_defined?(:log)
-    define_method("log") { $log }
-  end
-
   def configure(conf)
-    if conf['path']
-      if conf['path'].index('%S')
-        conf['time_slice_format'] = '%Y%m%d%H%M%S'
-      elsif conf['path'].index('%M')
-        conf['time_slice_format'] = '%Y%m%d%H%M'
-      elsif conf['path'].index('%H')
-        conf['time_slice_format'] = '%Y%m%d%H'
-      end
-    end
+    conf["time_slice_format"] = case conf["path"]
```

Comment:
The way of v0.14 API is to set `<buffer> timekey </buffer>` if `arg` of `<buffer>` includes "time" and `timekey` is not set.